src/backend/Pg.cpp

```diff
@@ -311,7 +570,9 @@ Pg::clear()
 
 //-----------------------------------------------------------------------------
 
-PgPool::PgPool(boost::json::object const& config)
+PgPool::PgPool(
+    boost::asio::io_context& ioc,
+    boost::json::object const& config) : ioc_(ioc)
```

Comment:
I'm actually looking at the code already implemented, and it looks like it forces the user to connect with username, password, and database name. In reporting mode, if the user does not specify these things, the connection defaults to using postgres on the local system as the same username/dbname as the rippled process.

Just a suggestion to make things easier for users running this on a single server.

Upon further thought, I'm not sure if asio would work with postgres on a local UNIX domain socket. So this requires the DB to be accessed over the network (even localhost), which requires specific configuration.